### PR TITLE
calibration of `TurbulenceConvection.jl` single-column model

### PR DESCRIPTION
**Pull request by haakon-e**
This PR implements an example for calibration of the `TurbulenceConvection.jl` single-column EDMF model.

### CODE REVIEW

**Comment by haakon-e** [comment]
@ilopezgp could you have a quick look at this?

I wanted to make a MVP for only calibration (without your other examples) into the CliMA EKP.jl repo. Can I remove the changes you made to `src/EnsembleKalmanProcess.jl` and `src/ParameterDistribution.jl` and possibly make that a separate PR later?

**Comment by charleskawczynski** [comment]
Do we want to check the data files into the repo? (e.g., the jld2 and .png files)

**Comment by haakon-e** [comment]
> Do we want to check the data files into the repo? (e.g., the jld2 and .png files)

I'm open to doing it either way, but the idea is to showcase expected results of running the example as-is.

**Comment by charleskawczynski** [comment]
> As an alternative, I could add the plots to the actual documentation?

I like that idea

File: examples/SCAMPy/experiments/scm_pycles_pipeline/calibrate.jl [hunk 1]
```diff
@@ -0,0 +1,223 @@
...
+        # Reference data specification
+        les_root = les_root,
+        les_name = "Bomex",
+        les_suffix = "aug09",
+        # Simulation case specification
+        scm_root = scm_root,
+        scm_name = "Bomex",
+        # Define observation window (s)
+        t_start = 4.0 * 3600,  # 4hrs
+        t_end = 24.0 * 3600,  # 6hrs
```
Comment (szy21) [inline examples/SCAMPy/experiments/scm_pycles_pipeline/calibrate.jl:59]:
```suggestion
        t_end = 24.0 * 3600,  # 24hrs
```

File: examples/SCAMPy/experiments/scm_pycles_pipeline/calibrate.jl [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+    # Calibrate using reference data and options described by the ReferenceModel struct.
+    ref_bomex = ReferenceModel(
+        # Define variables considered in the loss function
+        y_names = ["thetal_mean", "ql_mean", "qt_mean", "total_flux_h", "total_flux_qt"],
+        # Reference data specification
+        les_root = les_root,
+        les_name = "Bomex",
+        les_suffix = "aug09",
+        # Simulation case specification
+        scm_root = scm_root,
```
Comment (szy21) [inline examples/SCAMPy/experiments/scm_pycles_pipeline/calibrate.jl:55]: scm_root = pwd()?

File: docs/src/examples/TurbulenceConvection_example.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    "detrainment_factor"        => [bounded(0.0, 1.5*0.31)],
+)
+```
+as well as the names of the variables used in the loss function and location of reference data used to compute finite time-averaged statistics, which is stored in the `ReferenceModel` construct.
+
+Various flags allow us to easily decide whether to `perform_PCA` on the reference statistics, `normalize` the reference and forward model statistics, use LES or SCM data as reference (`model_type`), and restrict which type of output data is stored (`save_eki_data`, `save_ensemble_data`). 
+
+For the inversion step, we also specify the number of ensemble members `N_ens` and the number of iterations of the inversion method `N_iter`.
+
+Several methods and structures that enamble computation of the forward model, retrieval of time-averaged statistics or performing PCA is stored in the `helper_funcs.jl` file in the `src` folder of the `SCM` example directory.
```
Comment (szy21) [inline docs/src/examples/TurbulenceConvection_example.md:32]: ``
```suggestion
Several methods and structures that enable computation of the forward model, retrieval of time-averaged statistics or performing PCA is stored in the `helper_funcs.jl` file in the `src` folder of the `SCM` example directory.
```

**Review by costachris — APPROVED** [review]

File: examples/SCM/src/helper_funcs.jl [hunk 4]
```diff
@@ -0,0 +1,524 @@
+using NCDatasets
```
Comment (ilopezgp) [inline examples/SCM/src/helper_funcs.jl:1]: Many of the tools implemented in this PR are still being tested, and therefore should not be in the main branch of a published package. In addition, EKP.jl should be a generic optimizer and have lightweight examples. This PR is well out of that scope, since the example requires a lot of extra source code. Now that EKP.jl has been published as a package, let's move all this to another package within CliMA where we can have all the tools we need to use EKP to train the EDMF from LES/observations.

**Review by ilopezgp — CHANGES_REQUESTED** [review]
